corehq/motech/models.py

```diff
@@ -68,23 +93,103 @@ def plaintext_client_secret(self, plaintext):
             ciphertext=b64_aes_encrypt(plaintext)
             self.client_secret = f'${ALGO_AES}${ciphertext}'
 
+    @property
+    def last_token(self) -> Optional[dict]:
+        if self.last_token_aes:
+            plaintext = b64_aes_decrypt(self.last_token_aes)
+            return json.loads(plaintext)
+        return None
+
+    @last_token.setter
+    def last_token(self, token: Optional[dict]):
+        if token is None:
+            self.last_token_aes = ''
+        else:
+            plaintext = json.dumps(token)
+            self.last_token_aes = b64_aes_encrypt(plaintext)
+
     @property
     def notify_addresses(self):
         return [addr for addr in re.split('[, ]+', self.notify_addresses_str) if addr]
 
     def get_requests(self, payload_id, logger):
         from corehq.motech.requests import Requests
-        return Requests(
-            self.domain,
-            self.url,
-            self.username,
-            self.plaintext_password,
-            verify=not self.skip_cert_verify,
-            notify_addresses=self.notify_addresses,
-            payload_id=payload_id,
-            logger=logger,
-            auth_type=self.auth_type,
-        )
+
+        if not self._requests:
+            auth_manager = self.get_auth_manager()
+            self._requests = Requests(
+                self.domain,
+                self.url,
+                verify=not self.skip_cert_verify,
+                auth_manager=auth_manager,
+                notify_addresses=self.notify_addresses,
+                payload_id=payload_id,
+                logger=logger,
+            )
+        return self._requests
```

Comment:
nit: could you use `@memoized` instead here?  I see `_requests` is referenced elsewhere in the class, but I don't think that would preclude this usage here - calling `get_requests` doesn't necessarily mean a request has been made.

Reply:
Actually I've dropped this, because this code (and memoized) caches `payload_id`, which may be different the next time the cached instance is used. I considered using quickcache instead so that `payload_id` could be part of the key, but `logger` might not be hashable. In the end I thought maybe instantiating a Requests instance isn't so expensive, and I reuse instances where I can already.